### PR TITLE
Update routing_product.py

### LINKED ISSUE #354: Issues with rvx files generation for sub-basins with lake HRUs
* RavenPy version: 0.13
* Python version: 3.10.13
* Operating System: PAVICS, alpha test version server

### Description

I am creating rvx files from a shapefile, using the BasinMakerExtractor function. My shapefile contains lake HRUs. In these sub-basins, the river slope, river length, floodplain and river channel Manning's coefficient, max and min DEM are all set to -1.2345, following BasinMaker 'standards'. 

In the BasinMaker code, these sub-basins are handled differently when generating the .rvh and .rvp files. In the .rvh file: the REACH_LENGTH is set to ZERO-. In the .rvp file: the channel profile for these sub-basins is created using default Manning values. See the BasinMaker code starting at the line: https://github.com/dustming/basinmaker/blob/ad3bfbec167f8f6afa0cbb39f13611bf6e5e6838/basinmaker/hymodin/raveninput.py#L1420 

In RavenPy, these sub-basins are not handled differently (as far as I can see from the code), and certain key variables in the .rvh and .rvp files are therefore negative (-1.2345, with possible unit changes), causing issues when running a model (see example below).

@huard, would it be possible for us to add some code to the RavenPy BasinMakerExtractor to handle these cases the way it is done in BasinMaker?

### What I Did

When I try to run my model with these files:
```
outputs_path = run(modelname="SLSO_lowres_gr4j", configdir="/notebook_dir/INPACT-share/HBVECMOD/test_with_Aida_config/lowres2/")
```

The following error comes up: 

RavenError: Config directory: /notebook_dir/INPACT-share/HBVECMOD/test_with_Aida_config/lowres2
ParseClassPropertiesFile: Mannings n values in :RoughnessZones command must be greater than zero
ParseClassPropertiesFile: Mannings n values in :RoughnessZones command must be greater than zero
ParseClassPropertiesFile: Mannings n values in :RoughnessZones command must be greater than zero
ParseClassPropertiesFile: Mannings n values in :RoughnessZones command must be greater than zero
ParseClassPropertiesFile: Mannings n values in :RoughnessZones command must be greater than zero
CChannelXSect::Constructor: invalid mannings n

### PR DESCRIPTION
**Pull request by lou-a**
Modified routing_product.py to correctly set the river length to zero and set default values for reach attributes in sub-basins with no channel routing (i.e., sub-basins with lakes or headwater basins).

### Pull Request Checklist:
- [X] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes #354
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] CHANGELOG.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* ...

### CODE REVIEW

File: src/ravenpy/extractors/routing_product.py [hunk 1]
```diff
@@ -192,15 +198,24 @@ def _extract_reservoir(row) -> dict:
...
+            channel_elev = row["MeanElev"]
+            floodn = row["FloodP_n"]
+            channeln = row["Ch_n"]
+        else:
+            slope = 0.12345
+            channeln = 0.12345
+            floodn = 0.12345
+            channel_width = 0.12345
+            channel_depth = 0.12345
+            channel_elev = row["MeanElev"]
```
Comment (huard) [inline src/ravenpy/extractors/routing_product.py:218]:
```suggestion
        channel_elev = row["MeanElev"]
        if row["RivLength"] > 1.0:
            slope = max(row["RivSlope"], BasinMakerExtractor.MAX_RIVER_SLOPE)
            # SWAT: top width of channel when filled with water; bankfull width W_bnkfull
            channel_width = max(row["BkfWidth"], 1)
            # SWAT: depth of water in channel when filled to top of bank
            channel_depth = max(row["BkfDepth"], 1)
            floodn = row["FloodP_n"]
            channeln = row["Ch_n"]
        else:
            slope = 0.12345
            channeln = 0.12345
            floodn = 0.12345
            channel_width = 0.12345
            channel_depth = 0.12345
```

**Review by huard — APPROVED** [review]
Just add a note in the CHANGELOG, and a test to confirm everything is working as you expect.

**Comment by lou-a** [comment]
> Just add a note in the CHANGELOG, and a test to confirm everything is working as you expect.

@huard, let me know whether the tests seem ok. It's my first time coding tests 🤓

**Review by huard — APPROVED** [review]
looks good,

**Pull request merged**